Sliding window/LongestSubstringWithoutRepeatingCharacters3.py

```diff
@@ -0,0 +1,47 @@
+'''
+3. Longest Substring Without Repeating Characters
+Medium
+
+Given a string s, find the length of the longest 
+substring without repeating characters.
+
+Example 1:
+
+Input: s = "abcabcbb"
+Output: 3
+Explanation: The answer is "abc", with the length of 3.
+Example 2:
+
+Input: s = "bbbbb"
+Output: 1
+Explanation: The answer is "b", with the length of 1.
+Example 3:
+
+Input: s = "pwwkew"
+Output: 3
+Explanation: The answer is "wke", with the length of 3.
+Notice that the answer must be a substring, "pwke" is a subsequence and not a substring.
+ 
+
+Constraints:
+
+0 <= s.length <= 5 * 104
+s consists of English letters, digits, symbols and spaces.
+'''
+class Solution:
+    def lengthOfLongestSubstring(self, s: str) -> int:
+        charSet = set()
+        left = 0
+        result = 0
+        for right in range(len(s)):
+            while s[right] in charSet:
+                charSet.remove(s[left])
+                left += 1
+            charSet.add(s[right])
+            result = max(result, right - left + 1)
+        return result
```

Comment:
Solution is correct. The minor thing I'd mention is the set is not the best option here because it does not preserve the order. If you have a result string with all possible characters and the next one is the last in this string, you will need to remove all this characters one by one from the set. Instead, you can use a list or an OrderedSet, which is even better.